src/eduid_webapp/reset_password/views/reset_password.py

```diff
@@ -222,52 +222,57 @@ def __init__(self, msg):
         self.msg = msg
 
 
-TResetPasswordWithCodeSchemaSubclass = TypeVar(
-    'TResetPasswordWithCodeSchemaSubclass', bound=ResetPasswordWithCodeSchema
-)
-
+@dataclass
+class ResetContext(object):
+    state: Union[ResetPasswordEmailState, ResetPasswordEmailAndPhoneState]
+    user: User
+    error: Optional[Dict[str, Any]] = None  # If this is not None, the request should be aborted with this as result
 
-def _get_state_and_data(
-    schema_class: Type[TResetPasswordWithCodeSchemaSubclass],
-) -> Tuple[User, Union[ResetPasswordEmailState, ResetPasswordEmailAndPhoneState], Mapping[str, Any]]:
 
-    if not request.data:
-        raise BadStateOrData(ResetPwMsg.chpass_no_data)
-
-    data = json.loads(request.data)
+def _load_data(code: str, password: str) -> ResetContext:
+    """
+    Use a code to load reset-password state from the database.
 
-    if 'code' not in data:
-        raise BadStateOrData(ResetPwMsg.state_no_key)
+    Also validates the supplied password to make sure it conforms to the eduID service
+    requirements.
 
-    code = data['code']
+    :param code: User supplied password reset code
+    :param password: User supplied new password candidate
+    :return: ResetContext instance
+    """
     try:
         state = get_pwreset_state(code)
     except BadCode as e:
-        raise BadStateOrData(e.msg)
+        return ResetContext(
+            state=None,  # type: ignore
+            user=None,  # type: ignore
+            error=error_message(e.msg),
```

Comment:
I wonder if it is not better to set error to the enum value string and use error_message() when returning errors from views?